dune data provider

sampolgar opened the pull request:
Dune Analytics data provider as per the issue [here](https://github.com/sismo-core/sismo-hub/issues/1252)
I included the interface schema for the factory but since dune requires the API key, I decided not to update the `group-generators/helpers/data-providers/index.ts` file.
Once this is live I can submit 2 group generators to demonstrate how each query works

MartinGbz commented on group-generators/helpers/data-providers/dune/provider.ts:
```diff
@@ -0,0 +1,179 @@
+    );
+
+    console.log(
+      `\n dune_execution_id is ${execution_id} intial state is ${state} \n depending on size of query and Dune response time, this can take max 30 mins \n`
```
"intial" => "initial"

MartinGbz commented on group-generators/helpers/data-providers/dune/provider.ts:
```diff
@@ -0,0 +1,179 @@
+    );
+
+    console.log(
+      `\n dune_execution_id is ${execution_id} intial state is ${state} \n depending on size of query and Dune response time, this can take max 30 mins \n`
```
same here: "intial" => "initial"

MartinGbz commented on group-generators/helpers/data-providers/dune/provider.ts:
```diff
@@ -0,0 +1,179 @@
+   * @param queryParams optional! e.g. {address: '0x3CAaE25Ee616f2C8E13C74dA0813402eae3F496b', blocknumber: '66257668', chain: 'arbitrum'} for the Dune Query https://dune.com/queries/1618116
+   * @returns The FetchedData object in the <FetchedData> Type
+   */
+  public async executeQuery(
```
Maybe you can put only `executeQuery()` and `executeQueryCount()` in public, and all the other methods in private

MartinGbz commented on group-generators/helpers/data-providers/dune/provider.ts:
```diff
@@ -0,0 +1,179 @@
+import { DuneErrorFactory } from './errors';
```
In order to use the Data Provider in group generators you need to import it into the DataProvider index (group-generators/helpers/data-providers/index.ts)

- Import the DuneProvider
- Add the `DuneProvider` to the `dataProviders` object

MartinGbz commented on group-generators/helpers/data-providers/dune/interface-schema.json:
```diff
@@ -0,0 +1,63 @@
+{
```
Same here:

In order to use the DataProvider in the [Factory](https://factory.sismo.io/), you need to import it into the DataProvider index (group-generators/helpers/data-providers/index.ts)

- Import the `duneInterfaceSchema`
- Add the `duneInterfaceSchema` to the `dataProvidersInterfacesSchemas` object
- Add your count function to the `dataProvidersAPIEndpoints`
```
DuneProvider: {
  executeQueryCount: async (_: any) =>
  new DuneProvider().executeQueryCount(_),
},
...
```

sampolgar replied:
Thanks! 
How can we insert the API key when using the factory? The Dune query won't run without it.
Could we add it as an argument for the factory users?

leosayous21 replied:
Hey, we need to add it in the "infra" part like for the GITHUB_TOKEN
for instance you have it here in the workflows https://github.com/sismo-core/sismo-hub/blob/main/.github/workflows/deploy_prod.yml#L47 
and then it goes into the github action, then the serverless yaml https://github.com/sismo-core/sismo-hub/blob/main/serverless.yaml 
and env vars management into the sismo-hub :-) 

We can handle this part if you prefer

sampolgar replied:
Thanks, I can update the workflows folder. 
Looks like there's 2 naming conventions, 
`github-token: ${{ secrets.SH_GITHUB_TOKEN }}`
and
`transpose-api-key: ${{ secrets.TRANSPOSE_API_KEY }}`

Do you want the `SH_` before the key?

leosayous21 replied:
`DUNE_API_KEY` is better. the `SH_GITHUB_TOKEN` is due to a naming conflict in the github action auto injected GITHUB_TOKEN

leosayous21 commented on .github/workflows/check-new-groups-generation.yml:
```diff
@@ -22,6 +22,8 @@ jobs:
           DECENTRALIZED_SUBGRAPH_API_KEY: ${{ secrets.DECENTRALIZED_SUBGRAPH_API_KEY }} 
           GOOGLE_APPLICATION_CREDENTIALS:  ${{ secrets.GOOGLE_APPLICATION_CREDENTIALS }}
           SH_IGNORE_RESOLVING_ERRORS: "false"
+          dune-api-key: ${{ secrets.SH_DUNE_API_KEY }}
```
Hey @sampolgar It seems to me there is typo here "dune-api-key" it should be "SH_DUNE_API_KEY"

leosayous21 commented on .github/workflows/check-new-groups-generation.yml:
```diff
@@ -22,6 +22,8 @@ jobs:
           GOOGLE_APPLICATION_CREDENTIALS:  ${{ secrets.GOOGLE_APPLICATION_CREDENTIALS }}
           SH_IGNORE_RESOLVING_ERRORS: "false"
+          dune-api-key: ${{ secrets.SH_DUNE_API_KEY }}
+          SH_DUNE_API_KEY: ${{ secrets.HIVE_API_KEY }}
```
Here it should be "SH_DUNE_API_KEY"

leosayous21 reviewed (commented):
Hey @sampolgar could you use "DUNE_API_KEY" everywhere instead of "SH_DUNE_API_KEY"?

sampolgar commented:
Sorry for the secrets mix up, that's now fixed.
I updated the function arguments to receive a JSON object and typecast it to the Query Parameter Type.
Could you please check if that looks ok?

MartinGbz commented:
Hey @sampolgar there is a lot of confilcts due to your last commit, and because there were a lot of new data providers (ETHPorto reasons). Can you maybe revert your commit, merge main? (and if you want do the alphabetised fix again if you want, but it's not mandatory of course)

sampolgar commented:
@MartinGbz yes, can do.
The alphabetised fix made it faster to check where I'd put things, I'll do it again.
